imap_processing/utils.py

```diff
@@ -411,3 +323,22 @@ def packet_file_to_datasets(
         dataset_by_apid[apid] = ds
 
     return dataset_by_apid
+
+
+def convert_to_binary(bytes_data: list[bytes]) -> str:
+    """
+    Convert a list of bytes to a single byte string.
+
+    Parameters
+    ----------
+    bytes_data : list[bytes]
+        List of bytes to convert.
+
+    Returns
+    -------
+    binary_data : str
+        The binary string data.
+    """
+    binary_data = "".join(f"{byte:08b}" for byte in bytes_data)  # type: ignore [str-bytes-safe]
+
+    return binary_data
```

Comment:
This still isn't the same as the changelog suggestion. Why does it accept list of bytes and not just bytes?
https://github.com/medley56/space_packet_parser/blob/502159d03b2f45e18ac0311f89d65108cd2a1f57/docs/source/changelog.md?plain=1#L32

Also, this should be named "binary_string" or something with string in it because binary is what we are coming _from_ not what we are going _to_

```suggestion
def convert_to_binary_string(data: bytes) -> str:
    """
    Convert bytes to a string representation.

    Parameters
    ----------
    data : bytes
        Bytes to convert

    Returns
    -------
    : str
        The binary data as a string
    """
    return f"{int.from_bytes(data, byteorder='big'):0{len(data)*8}b}"
```

Reply:
Thank you! I did remember seeing Gavin's note but totally forgot when I refactored it.

Reply:
To you list question, good catch. I was in two places when I wrote that.